### PR TITLE
Serializer with type information

### PR DESCRIPTION
**Pull request by hg-ms**

### CODE REVIEW

File: persistence/binary/src/main/java/one/microstream/persistence/binary/util/PersistenceExceptionTypeImportTypeNotFound.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * available at https://www.gnu.org/software/classpath/license.html.
+ * 
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ * #L%
+ */
+
+import one.microstream.persistence.exceptions.PersistenceException;
+
+@SuppressWarnings("serial")
+public class PersistenceExceptionTypeImportTypeNotFound extends PersistenceException
```
Comment (fh-ms) [inline persistence/binary/src/main/java/one/microstream/persistence/binary/util/PersistenceExceptionTypeImportTypeNotFound.java]: Should be called `BineraPersistenceExceptionTypeImportTypeNotFound`, extends from `BinaryPersistenceException` and be in package `one.microstream.persistence.binary.exceptions`.

File: persistence/binary/src/main/java/one/microstream/persistence/binary/util/SerializerTypeInfo.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ * with the GNU Classpath Exception which is
+ * available at https://www.gnu.org/software/classpath/license.html.
+ * 
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ * #L%
+ */
+
+import java.util.Collection;
+
+public class SerializerTypeInfo
```
Comment (fh-ms) [inline persistence/binary/src/main/java/one/microstream/persistence/binary/util/SerializerTypeInfo.java:25]: Better work with our collections than java.util.Collection and arrays. Array member assignments are mutable.
Best to hand in a `XGettingList<String>` and assign it to a `XGettingList<String>` field by calling the `immure` method to make it immutable.

**Review by fh-ms — CHANGES_REQUESTED** [review]

File: persistence/binary/src/main/java/one/microstream/persistence/binary/util/PersistenceExceptionTypeImportTypeNotFound.java (hunk 1 above)
Comment (hg-ms) [reply persistence/binary/src/main/java/one/microstream/persistence/binary/util/PersistenceExceptionTypeImportTypeNotFound.java]: ok, changed

File: persistence/binary/src/main/java/one/microstream/persistence/binary/util/SerializerTypeInfo.java (hunk 2 above)
Comment (hg-ms) [reply persistence/binary/src/main/java/one/microstream/persistence/binary/util/SerializerTypeInfo.java:25]: ok, changed

**Review by fh-ms — APPROVED** [review]

**Pull request merged**